src/PrestaShopBundle/Form/Admin/Sell/Order/Invoices/InvoiceOptionsDataProvider.php

```diff
@@ -70,33 +62,6 @@ public function getData()
      */
     public function setData(array $data)
     {
-        if ($errors = $this->validate($data)) {
-            return $errors;
-        }
-
         return $this->invoiceOptionsConfiguration->updateConfiguration($data);
     }
-
-    /**
-     * Perform validations on form data.
-     *
-     * @param array $data
-     *
-     * @return array Array of errors if any
-     */
-    private function validate(array $data)
```

Comment:
I dont know if it's good to remove this validation. 🤔  indeed it's a duplicate validation with what the Form does. But these are 2 classes, a module developer could choose to replace the FormType by another implementation and use this one.

I'm fifty/fifty on this.

@PierreRambaud @jolelievre wdyt ? we have moved this validation to FormType using Constraints, shall we keep it nonetheless at this level or remove it?

Reply:
A good point that InvoiceOptionsDataProvider could be used not just with formType. 

Problem with validation in dataProvider is that it's harder to affect it by the module. Hook is no longer enough and you need to decorate the service, but not having validation here means data provider can't be reliably used outside the FormType. 

Also now that you mentioned it, it makes sense DataProvider should be responsible for validating what it saves not the formType. Right now I would be for having validation both in constraints if possible and in DataProvider, even if it makes harder to remove certain validation for module developers.